### PR TITLE
Add ?storage arg to mount API

### PR DESCRIPTION
**Pull request by DaveCTurner**
This commit introduces a new flag, `?storage`, controlling the storage
of the local copy of a searchable snapshot.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/2 (unrelated failure, reproduces without these changes)

File: x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsSystemIndicesIntegTests.java [hunk 1]
```diff
@@ -72,7 +72,8 @@ private void executeTest(final String indexName, final Client client) throws Exc
             indexName,
             Settings.builder().put(IndexMetadata.SETTING_INDEX_HIDDEN, randomBoolean()).build(),
             Strings.EMPTY_ARRAY,
-            true
+            true,
+            randomFrom(MountSearchableSnapshotRequest.Storage.FULL_COPY, MountSearchableSnapshotRequest.Storage.SHARED_CACHE)
```
Comment (ywelsch) [inline x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsSystemIndicesIntegTests.java:76]: just `randomFrom(Storage.values())`?

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/2

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/searchablesnapshots/MountSearchableSnapshotRequest.java [hunk 2]
```diff
@@ -162,6 +186,13 @@ public Settings indexSettings() {
         return ignoredIndexSettings;
     }
 
+    /**
+     * @return whether the local copy of the snapshot is partial ({@code true}) or complete ({@code false}).
```
Comment (tlrx) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/searchablesnapshots/MountSearchableSnapshotRequest.java:190]: Maybe drop (or adjust the doc) here?

**Review by tlrx — APPROVED** [review]
LGTM 2

**Comment by DaveCTurner** [comment]
YOLO merge to the feature branch, a PR to master is impending.

**Pull request merged**